core/src/impl/Kokkos_HostSpace_deepcopy.hpp

```diff
@@ -28,9 +28,9 @@ void hostspace_fence(const DefaultHostExecutionSpace& exec);
 void hostspace_parallel_deepcopy(void* dst, const void* src, ptrdiff_t n);
 // DeepCopy called with an execution space that can't access HostSpace
 void hostspace_parallel_deepcopy_async(void* dst, const void* src, ptrdiff_t n);
-void hostspace_parallel_deepcopy_async(const DefaultHostExecutionSpace& exec,
-                                       void* dst, const void* src, ptrdiff_t n);
-
+template <typename ExecutionSpace>
```

Comment:
Same question here:  is there a reason to use `typename` vs. `class` in templated expression?